### DISCUSSION TITLE
Why is concourse not using the latest version of the resource?

### DISCUSSION THREAD

**Discussion by patmaddox** [concourse/concourse#7784] [Help & Support]
We're encountering something strange that I _think_ is new after upgrading from 7.4.0 (or maybe 7.5.0) to 7.6.0. We're using the helm chart.

Concourse builds with an old version of a resource. I can't figure out why.

When I check the resource, it succeeds and shows the current version as the git head (but it doesn't appear to create a new version - it's just the check that shows the correct SHA). But when I trigger a build, it uses an old SHA.

No versions are pinned or disabled. Concourse is just picking an old version to build. When I disable the old version and try to build, Concourse gets stuck on "waiting for new version of resource".

I have run `clear-resource-cache` multiple times. It just says `0 caches removed` and I still get the same behavior.

Has anyone seen anything like this? How do I get past it?

Edit: Even deleting the pipeline and re-creating entirely, it still uses an old version of the repo. So I guess something is being cached somewhere?

The resource check succeeds and sees the correct SHA:

> HEAD is now at 57bf0ed Add some docs (to try to get concourse to pick up a new version)

but doesn't make a new version for it:

![image](https://user-images.githubusercontent.com/1111/141032520-97fbefe8-efae-4131-a1b9-a1c81d6fcdc4.png)

**Comment by crsimmons** [comment]
Can you post an example pipeline where this happens? Most of the times I've seen errors like that it's some kind of passed constraint thing that prevents the job from "seeing" the intended version but it sounds like this might not be the case for you.

**Comment by taylorsilva** [comment]
7.6.0 came with git-resource v1.14.5
7.5.0 came with git-resource v1.14.4
7.4.0 came with git-resource v1.14.0

You could try using each of these different versions in your pipelines by adding this to your pipeline:
```yaml
resource_types:
- name: git
  type: registry-image
  source:
    repository: concourse/git-resource
    tag: 1.14.xxx #put the version you want to test against here
```
This will tell us if it's a problem with Concourse or the resource itself.

**Reply by patmaddox** [reply]
I can probably try that some time in the next week.

The other thing is that we have lots of the same resource defined across pipelines (`ratio-ops-git` provides a set of shared scripts), and around this time we introduced using different branches with the same resource name. I got around it by renaming resources to things like `project-a-ratio-ops-git-other-branch`. Global resources are enabled by default in the helm chart. I disabled global resources and saw the same problem.

**Comment by tomasgareau** [comment]
I've bumped into this a couple times and usually find that it's the `paths` or `ignore_paths` arguments of the `git` resource -- if you've specified `paths` and no files in that path have changed (or that path doesn't exist) then the `git-resource` won't emit a new version.

I found the easiest way to debug was to walk through the [`check`](https://github.com/concourse/git-resource/blob/master/assets/check) script and manually run the commands in there (in particular, the `list_command` argument that it builds up).

In my case, this turned out to be:

```
git rev-list --all --first-parent <YOUR_PATHS_ARGS_HERE> | git rev-list --stdin --date-order --no-walk=unsorted -1 --reverse | jq -R '.' | jq -s "map({ref: .})"
```
